### PR TITLE
[OTA-2123] Handle provisioning download errors

### PR DESCRIPTION
**Pull request by piotrsmol**
Signed-off-by: Piotr Smol <piotrsmol.pvt@gmail.com>

### CODE REVIEW

File: ota-plus-web/app/reactapp/src/components/profile/access-keys/ListItem.jsx [hunk 1]
```diff
@@ -2,28 +2,44 @@
...
+          {`Start date: ${validFrom.toDateString()}  ${validFrom.toLocaleTimeString()}`}
+        </div>
+        <div className='column'>
+          {`End date: ${validUntil.toDateString()} ${validUntil.toLocaleTimeString()}`}
+        </div>
+        <div className='column'>
+          <img 
+            src='/assets/img/icons/download_key.svg' 
+            className='download-key-link' 
+            id={'download-key-link-' + provisioningKey.description}
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp/src/components/profile/access-keys/ListItem.jsx]: Use string literals

File: ota-plus-web/app/reactapp/src/stores/ProvisioningStore.js [hunk 2]
```diff
@@ -2,9 +2,18 @@
...
+  API_PROVISIONING_STATUS, 
+  API_PROVISIONING_DOWNLOAD, 
+  API_PROVISIONING_DETAILS, 
+  API_PROVISIONING_KEYS_FETCH, 
+  API_PROVISIONING_KEY_CREATE, 
+  API_NAMESPACE_SETUP_STEPS 
+} from '../config';
 import { resetAsync, handleAsyncSuccess, handleAsyncError } from '../utils/Common';
 import _ from 'lodash';
+import { notification } from 'antd';
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp/src/stores/ProvisioningStore.js]: Module imports should be on top

File: ota-plus-web/app/reactapp/src/stores/ProvisioningStore.js [hunk 3]
```diff
@@ -148,6 +157,24 @@ export default class ProvisioningStore {
...
   }
 
+  downloadProvisioningKey = async (id) => {
+    const downloadUrl = `${API_PROVISIONING_DOWNLOAD}/${id}`;
+    try {
+      const { status } = await axios.get(downloadUrl);
+      if (status === HTTP_CODE_200_OK) {
+        location.href = downloadUrl;
+      } else {
+        throw new Error();
```
Comment (piotrcichonmee) [inline ota-plus-web/app/reactapp/src/stores/ProvisioningStore.js]: Maybe add here notification (the same as below) too?

File: ota-plus-web/app/reactapp/src/components/profile/access-keys/ListItem.jsx (hunk 1 above)
Comment (piotrsmol) [reply ota-plus-web/app/reactapp/src/components/profile/access-keys/ListItem.jsx]: done

File: ota-plus-web/app/reactapp/src/stores/ProvisioningStore.js (hunk 2 above)
Comment (piotrsmol) [reply ota-plus-web/app/reactapp/src/stores/ProvisioningStore.js]: done

**Pull request merged**